### PR TITLE
Override create_or_find_by methods and implement proper exception handling

### PR DESCRIPTION
**Pull request by alexeyfrank**

### CODE REVIEW

File: lib/database_validations/lib/validations.rb [hunk 1]
```diff
@@ -6,6 +6,28 @@ module Validations
...
+      def create_or_find_by(attributes, &block)
+        super
+      rescue ActiveRecord::RecordInvalid => e
+        rescue_create_or_find_by_uniqueness_exception(e, attributes)
+      end
+
+      private
+
+      def rescue_create_or_find_by_uniqueness_exception(err, attributes)
+        raise err unless err.is_a?(ActiveRecord::NotUnique) || err&.cause&.is_a?(ActiveRecord::RecordNotUnique)
```
Comment (djezzzl) [inline lib/database_validations/lib/validations.rb:25]: **Question:** Do we need `err.is_a?(ActiveRecord::NotUnique)`? I thought it's processed in `super` already.

Comment (djezzzl) [reply lib/database_validations/lib/validations.rb:25]: **Note:**

Never knew about `cause` available on the ActiveRecord errors :+1: But do we need `err&`, won't it be not nil always in your case?

**Review by djezzzl — COMMENTED** [review]
Could you please add some tests? Otherwise, we don't really know if it's working properly.

File: lib/database_validations/lib/validations.rb [hunk 2]
```diff
@@ -6,6 +6,28 @@ module Validations
...
+      def create_or_find_by(attributes, &block)
+        super
+      rescue ActiveRecord::RecordInvalid => e
+        rescue_create_or_find_by_uniqueness_exception(e, attributes)
+      end
+
+      private
+
+      def rescue_create_or_find_by_uniqueness_exception(err, attributes)
+        raise err unless err.cause&.is_a?(ActiveRecord::RecordNotUnique)
```
Comment (ojab) [inline lib/database_validations/lib/validations.rb:25]: Maybe add this into `rescue_from_database_exceptions(validate, &block)` along with `ActiveRecord::InvalidForeignKey` handling?

**Pull request closed without merging**